### ISSUE TITLE
[BUG] zend_mm_heap corrupted / Segmentation fault: 11

### ISSUE THREAD

**Issue by blieb** [barbushin/php-imap#320]
**Environment (please complete the following information):**
 - PHP IMAP version: 3.0.14 / 3.0.15
 - PHP Version: 7.1.12 / 7.2.1
 - Type of execution: CLI

**Describe the bug**
Since PHP IMAP Version 3.0.14 I get error when I try to use getMail.

I've tried it on 2 php versions,
on php 7.1.12 I get the error: "zend_mm_heap corrupted"
on php 7.2.1 I get the error: "Segmentation fault: 11"

I'm running this on CLI on MacOS. Php versions from MAMP PRO.
PHP IMAP Version 3.0.13 works fine. 
**To Reproduce**
Steps to reproduce the behavior.

The used code:
```php
        $mailbox = new \PhpImap\Mailbox(
            $config['server'],
            $config['username'],
            $config['password']
        );
        $mailsIds = $mailbox->searchMailbox('ALL');
        $mail = $mailbox->getMail($mailIds[0]);
```

**Expected behavior**
Getting email

**Screenshots / Outputs**
WHen I run it I get on php 7.1 "zend_mm_heap corrupted"
I also tried it on php 7.2 but then I get error "Segmentation fault: 11"

**Comment by Sebi94nbg** [comment]
From which mail server do you try to get emails from? GMail / G Suite, Microsoft Exchange, Own?

Do you have installed and configured any kind of caching? On the CLI, you may want to disable caching in your `php.ini`: `opcache.enable_cli=0` (if opcache is used for example)

See https://stackoverflow.com/questions/2247977/what-does-zend-mm-heap-corrupted-mean.

**Comment by blieb** [comment]
Yes already played with that. but no success, I do not have caching enabled. 

The mailserver I use is an own mailserver. Running dovecot imap server. (directadmin server)

**Comment by Sebi94nbg** [comment]
My server is running with Dovecot 2.2.27, but I don't have these problems.

Can you please show your customized Dovecot configuration (only non-default settings)? `doveconf -an`

Then I can compare these settings with those from mine and check, what's causing the issue.

**Comment by blieb** [comment]
I also have this problem when I try to use gmail. So is not server specific.

I'm going to try it on an other system. Mac and linux.

**Comment by blieb** [comment]
I've running the code inside cakephp framework. Looks like it only fails in combination with that. I continue testing to see if I can find the reason.

**Comment by Sebi94nbg** [comment]
CakePHP also provides a caching solution: https://book.cakephp.org/3.0/en/core-libraries/caching.html

Is this maybe causing this issue?

**Comment by blieb** [comment]
Cakephp only caches data. and only when you ask it too.

I've created a new cakephp 3.7 project. Here an updated reproduce step by step:
`composer create-project --prefer-dist cakephp/app`

go to project folder (app)

`composer require php-imap/php-imap`

add file: src/Command/testCommand.php (create folder Command)
```php
<?php

namespace App\Command;
use Cake\Console\Arguments;
use Cake\Console\Command;
use Cake\Console\ConsoleIo;

class TestCommand extends Command {

    public function execute(Arguments $args, ConsoleIo $io)
    {
        $mailbox = new \PhpImap\Mailbox(
            '{imap.gmail.com:993/imap/ssl}INBOX', // IMAP server and mailbox folder
            '', // Username for the before configured mailbox
            '' // Password for the before configured username
        );
        $mailsIds = $mailbox->searchMailbox('ALL');
        foreach ($mailsIds as $mailId) {
            $mail = $mailbox->getMail($mailId);
            print_r($mail);
            exit;
        }
    }
}

```

(fill in username/password)

run command by running: `bin/cake test`

**Comment by blieb** [comment]
Looks like a problem with iconv. It crashes at this line:
https://github.com/barbushin/php-imap/blob/c0750ec4ce4e00be2e884005b2fbd94ccd64a34d/src/PhpImap/Mailbox.php#L1013

**Comment by Sebi94nbg** [comment]
With which error message does it crash?

Just remove the `@` sign in front of `iconv(`, if you don't see any error messages.

**Comment by Sebi94nbg** [comment]
@blieb I've changed a few things. Can you please test it again?
- Commit https://github.com/barbushin/php-imap/commit/f2cb352e1a802e9ba688b8e074c5c64405ac08b9
- Commit https://github.com/barbushin/php-imap/commit/caee2a668fc0d82275dc166b48a55361a24778f9

Use the current code of the `develop` branch:
`composer require php-imap/php-imap:dev-develop`

**Comment by blieb** [comment]
Yes the current dev version works without problems

**Comment by Sebi94nbg** [comment]
Great! These changes are already included in the latest release [v3.0.16](https://github.com/barbushin/php-imap/releases/tag/3.0.16). :)
